docs/software-architecture-and-design.md

```diff
@@ -331,29 +332,33 @@ Each component is described in more detail in their own sections.
 
 
 # Backend
-The `Backend` is responsible for all communication with the "outside world". The responsibilities of the `Backend` can be broken down into Input and Output.
+The `Backend` is responsible for all communication with the "outside world". The responsibilities of the `Backend` can be broken down into communication with `SensorProto` and [Primitives](#primitives) messages:
```

Comment:
"communication with SensorProto and Primitives" doesn't quite sound right to me, since SensorProto is a message type, not a "thing" that the Backend communicates with. Maybe "broken down into sending SensorProto and receiving Primitives"?

Reply:
This doesn't include the responsibility of communicating with the "outside world" (which I know is mentioned below but seems like it would fit here)

Reply:
I wanted to describe it in terms of the messages it _uses_. I updated it to reflect that